compiler/noirc_evaluator/src/ssa/ir/integer.rs

```diff
@@ -0,0 +1,122 @@
+        }
+    }
+
+    /// Convert back into a field.
+    pub(crate) fn into_numeric_constant(self) -> (FieldElement, NumericType) {
+        match self {
+            Self::Signed { value, bit_size } => (
+                binary::convert_signed_integer_to_field_element(value, bit_size),
+                NumericType::signed(bit_size),
+            ),
+            Self::Unsigned { value, bit_size } => {
+                (FieldElement::from(value), NumericType::unsigned(bit_size))
+            }
+        }
+    }
+
+    /// Reduce two constants into a result by applying functions on them if their signedness matches.
+    pub(crate) fn reduce<T>(
+        self,
+        other: Self,
+        s: impl Fn(i128, i128) -> T,
+        u: impl Fn(u128, u128) -> T,
+    ) -> Option<T> {
+        match (self, other) {
+            (Self::Signed { value: a, .. }, Self::Signed { value: b, .. }) => Some(s(a, b)),
+            (Self::Unsigned { value: a, .. }, Self::Unsigned { value: b, .. }) => Some(u(a, b)),
+            _ => None,
+        }
+    }
+
+    /// Apply functions on signed/unsigned values.
+    pub(crate) fn apply<T>(&self, s: impl Fn(i128) -> T, u: impl Fn(u128) -> T) -> T {
+        match self {
+            Self::Signed { value, .. } => s(*value),
+            Self::Unsigned { value, .. } => u(*value),
+        }
+    }
+
+    /// Increment the value by 1, saturating at the maximum value.
+    pub(crate) fn inc(self) -> Self {
+        match self {
+            Self::Signed { value, bit_size } => {
+                Self::Signed { value: value.saturating_add(1), bit_size }
+            }
+            Self::Unsigned { value, bit_size } => {
+                Self::Unsigned { value: value.saturating_add(1), bit_size }
+            }
+        }
+    }
+
+    /// Decrement the value by 1, saturating at the minimum value.
+    pub(crate) fn dec(self) -> Self {
+        match self {
+            Self::Signed { value, bit_size } => {
+                Self::Signed { value: value.saturating_sub(1), bit_size }
+            }
+            Self::Unsigned { value, bit_size } => {
+                Self::Unsigned { value: value.saturating_sub(1), bit_size }
+            }
+        }
+    }
+
+    pub(crate) fn is_zero(&self) -> bool {
+        match self {
+            Self::Signed { value, .. } => value.is_zero(),
+            Self::Unsigned { value, .. } => value.is_zero(),
+        }
+    }
+}
+
+impl PartialOrd for IntegerConstant {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        match (self, other) {
+            (Self::Signed { value: a, .. }, Self::Signed { value: b, .. }) => a.partial_cmp(b),
+            (Self::Signed { value: a, .. }, Self::Unsigned { value: b, .. }) => {
+                if a.is_negative() || *b > i128::MAX as u128 {
+                    Some(Ordering::Less)
+                } else {
+                    let a: u128 = (*a).try_into().ok()?;
+                    a.partial_cmp(b)
+                }
```

Comment:
```suggestion
                if a.is_negative() {
                    Some(Ordering::Less)
                } else {
                    (*a).try_into().ok().and_then(|a: u128| a.partial_cmp(b))                
                }
```
nit: we should be able to simplify this if statement